runtime/core/exec_aten/util/scalar_type_util.h

```diff
@@ -979,12 +935,97 @@ struct promote_types {
   ET_INTERNAL_SWITCH_CASE(                                             \
       ::executorch::aten::ScalarType::Int, CTYPE_ALIAS, __VA_ARGS__)   \
   ET_INTERNAL_SWITCH_CASE(                                             \
-      ::executorch::aten::ScalarType::Long, CTYPE_ALIAS, __VA_ARGS__)  \
+      ::executorch::aten::ScalarType::Long, CTYPE_ALIAS, __VA_ARGS__)
+
+#define ET_INTERNAL_SWITCH_CASE_UINT_TYPES(CTYPE_ALIAS, ...)            \
+  ET_INTERNAL_SWITCH_CASE(                                              \
+      ::executorch::aten::ScalarType::UInt16, CTYPE_ALIAS, __VA_ARGS__) \
+  ET_INTERNAL_SWITCH_CASE(                                              \
+      ::executorch::aten::ScalarType::UInt32, CTYPE_ALIAS, __VA_ARGS__) \
+  ET_INTERNAL_SWITCH_CASE(                                              \
+      ::executorch::aten::ScalarType::UInt64, CTYPE_ALIAS, __VA_ARGS__)
+
+#define ET_INTERNAL_SWITCH_CASE_ALLINT_TYPES(CTYPE_ALIAS, ...) \
+  ET_INTERNAL_SWITCH_CASE_INT_TYPES(CTYPE_ALIAS, __VA_ARGS__)  \
+  ET_INTERNAL_SWITCH_CASE_UINT_TYPES(CTYPE_ALIAS, __VA_ARGS__)
+
+#define ET_INTERNAL_SWITCH_CASE_ALLINTB_TYPES(CTYPE_ALIAS, ...)  \
+  ET_INTERNAL_SWITCH_CASE_ALLINT_TYPES(CTYPE_ALIAS, __VA_ARGS__) \
+  ET_INTERNAL_SWITCH_CASE_BOOL_TYPES(CTYPE_ALIAS, __VA_ARGS__)
+
+#define ET_INTERNAL_SWITCH_CASE_FLOAT_TYPES(CTYPE_ALIAS, ...)          \
   ET_INTERNAL_SWITCH_CASE(                                             \
       ::executorch::aten::ScalarType::Float, CTYPE_ALIAS, __VA_ARGS__) \
   ET_INTERNAL_SWITCH_CASE(                                             \
       ::executorch::aten::ScalarType::Double, CTYPE_ALIAS, __VA_ARGS__)
 
+#define ET_INTERNAL_SWITCH_CASE_ALLFLOAT_TYPES(CTYPE_ALIAS, ...)      \
```

Comment:
having these both exist is confusing -- "Half and BFloat16 are part of all float types, but they're not float types"?

Reply:
Half and BFloat16 are not part of the FLOAT set today. That's why we have FLOATH and FLOATHBF16.
I am using the prefix "ALL" to denote the set containing all of the types belonging to a given category (INT, FLOAT, REAL, COMPLEX, NUMERIC)